### PR TITLE
Register learning rate schedulers with app_context and unittests

### PR DESCRIPTION
**Pull request by kparichay**
- register learning rate scheduler with the app context and regiter its
type factories.
- Add unittests for the constant and exponential learning rate schedulers.

See Also #1776

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

**Review by zhoonit — APPROVED** [review]
LGTM :)

File: test/unittest/unittest_nntrainer_lr_scheduler.cpp [hunk 1]
```diff
@@ -87,6 +87,31 @@ TEST(lr_constant, prop_03_p) {
...
+/**
+ * @brief test set and get learning rate
+ *
+ */
+TEST(lr_constant, final_02_p) {
+  auto &ac = nntrainer::AppContext::Global();
+  auto lr = ac.createObject<nntrainer::LearningRateScheduler>("constant");
+
+  EXPECT_NO_THROW(lr->setProperty({"learning_rate=1.0"}));
+  /** fails as learning rate property is not set */
```
Comment (lhs8928) [inline test/unittest/unittest_nntrainer_lr_scheduler.cpp:111]: Please remove this

**Review by lhs8928 — APPROVED** [review]
LGTM

File: test/unittest/unittest_nntrainer_lr_scheduler.cpp (hunk 1 above)
Comment (kparichay) [reply test/unittest/unittest_nntrainer_lr_scheduler.cpp]: done.
